File: Arrays and Hashing/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        n = len(s) // 2
        compare = ""
        compare2 = ""
        for i in s[:n]:
            compare += i.lower()


        for i in range(-1, -n, -1):
            compare2 += s[i]


        print(compare)
        print(compare2)
        return compare == compare2
    

        

solution = Solution()
print(solution.isPalindrome("Was it a car or a cat I saw?"))

File: Two Pointers/ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase().replace(" ", "");
        int l = 0;
        int r = s.length() - 1;
        System.out.println(s);
        while (l < r){
            if (s.charAt(l) != s.charAt(r)){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.isPalindrome("Was it a car or a cat I saw?"));
    }
}
